integrations/appboy/lib/index.js

```diff
@@ -356,6 +367,14 @@ Appboy.prototype.track = function(track) {
     delete properties[key];
   }, reserved);
 
+  // Remove nested objects as Braze doesn't support nested objects in tracking calls
+  // https://segment.com/docs/destinations/braze/#track
+  each(function(value, key) {
+    if (val != null && typeof value === 'object') {
```

Comment:
Whoops. This should actually be `value` instead of `val`. @NikoRoberts

Reply:
```suggestion
    if (value != null && typeof value === 'object') {
```

Reply:
Done